### DISCUSSION TITLE
how to get the amplitude image of O3X via Python codes

### DISCUSSION THREAD

**Discussion by heruoyu1206** [ifm/ifm3d#451] [Q&A]
with the guidance and Python code example here:
https://api.ifm3d.com/stable/examples/o3r/getting_data/getting_data.html

i can get the amplitude image from O3D camera, but not from O3X. And the error message is:
_Lib: A buffer with the requested buffer_id is not available.: buffer_id: 103_
So, how can i get the amplitude image from an O3X120 camera via Python codes?

thanks !

**Comment by desengph** [comment]
@heruoyu1206 We are currently on updating the O3X ifm3d API examples based on API versions >= 1.3.
This should only take a short amount of time.
We will keep you updated on the progress and any example code.

**Comment by heruoyu1206** [comment]
got it, thanks! 
it would be very nice if you can keep be informed via email. My email address is 

ruoyu.he@ifm.com. 

thank you!

**Comment by lola-masson** [comment]
Hi @heruoyu1206,

We have found an issue with the amplitude buffers for the O3X, where neither AMPLITUDE_IMAGE nor NORM_AMPLITUDE_IMAGE would return data. This is going to be fixed in the next release but in the meantime, a workaround is to request both buffers at the same time, and get the data from NORM_AMPLITUDE_IMAGE:
```python
from ifm3dpy.device import O3X
from ifm3dpy.framegrabber import FrameGrabber, buffer_id

o3x = O3X()
fg = FrameGrabber(o3x)
fg.start([buffer_id.AMPLITUDE_IMAGE, buffer_id.NORM_AMPLITUDE_IMAGE])
[ok, frame] = fg.wait_for_frame().wait_for(1000)
amp = frame.get_buffer(buffer_id.NORM_AMPLITUDE_IMAGE)
```
